datafusion/functions-table/src/generate_series.rs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::array::Int64Array;
+use arrow::datatypes::{DataType, Field, Schema, SchemaRef};
+use arrow::record_batch::RecordBatch;
+use async_trait::async_trait;
+use datafusion_catalog::Session;
+use datafusion_catalog::TableFunctionImpl;
+use datafusion_catalog::TableProvider;
+use datafusion_common::{not_impl_err, plan_err, Result, ScalarValue};
+use datafusion_expr::{Expr, TableType};
+use datafusion_physical_plan::memory::{StreamingBatchGenerator, StreamingMemoryExec};
+use datafusion_physical_plan::ExecutionPlan;
+use std::fmt;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+/// Table that generates a series of integers from `start`(inclusive) to `end`(inclusive)
+#[derive(Debug, Clone)]
+struct GenerateSeriesTable {
+    schema: SchemaRef,
+    // None if input is Null
+    start: Option<i64>,
+    // None if input is Null
+    end: Option<i64>,
+}
+
+/// Table state that generates a series of integers from `start`(inclusive) to `end`(inclusive)
+#[derive(Debug, Clone)]
+struct GenerateSeriesState {
+    schema: SchemaRef,
+    _start: i64,
```

Comment:
Is there any reason to keep this?

Reply:
This field is for display, i added a comment to explain